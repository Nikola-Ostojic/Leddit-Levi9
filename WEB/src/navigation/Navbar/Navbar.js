import React from "react";
import { NavLink, withRouter } from "react-router-dom";
import { AppBar, Toolbar, Button, Typography } from "@material-ui/core";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import { makeStyles } from "@material-ui/core/styles";

import styles from "./Navbar.module.css";

const useStyles = makeStyles({
  appBar: {
    // Making the AppBar be "above" Drawer
    zIndex: 1400
  },
  title: {
    flexGrow: 1,
    textAlign: "left"
  }
});
export const Navbar = ({ user, logout, setDrawerOpen, drawerOpen }) => {
  const classes = useStyles();
  return (
    <AppBar position="fixed" className={`${classes.appBar} ${styles.links}`}>
      <Toolbar>
        <IconButton
          edge="start"
          color="inherit"
          aria-label="menu"
          onClick={() => setDrawerOpen(!drawerOpen)}
        >
          <MenuIcon />
        </IconButton>
        <Typography variant="h6" className={classes.title}>
          Leddit
        </Typography>
        <Button color="inherit">
          {user ? (
            <NavLink to="/auth" onClick={logout}>
              <Typography>Logout {user}</Typography>
            </NavLink>
          ) : (
            <NavLink to="/auth">
              <Typography>Login</Typography>
            </NavLink>
          )}
        </Button>
      </Toolbar>
    </AppBar>
  );
};

export default withRouter(Navbar);
