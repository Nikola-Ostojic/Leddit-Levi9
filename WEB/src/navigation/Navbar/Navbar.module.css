.nav {
    justify-content: space-between;
}

.links a {
    margin-right: 10px;
    text-decoration: none !important;
    color: white;
    font-weight: bold;
    font-size: 18px;
}