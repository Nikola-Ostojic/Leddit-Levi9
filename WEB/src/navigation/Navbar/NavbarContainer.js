import { connect } from "react-redux";

import { getUsername } from "modules/Auth/AuthSelectors";
import { logout } from "modules/Auth/AuthActions";

import Navbar from "./Navbar";

export const mapStateToProps = (state) => ({
  user: getUsername(state)
});

export default connect(mapStateToProps, { logout })(Navbar);
