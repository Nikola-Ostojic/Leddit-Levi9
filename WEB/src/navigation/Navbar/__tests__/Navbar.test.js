import React from "react";
import { shallow } from "enzyme";

import { Navbar } from "../Navbar";

describe("Navbar tests", () => {
  const mockedProps = {
    user: "user",
    logout: jest.fn(),
    setDrawerOpen: jest.fn(),
    drawerOpen: false
  };
  it("Should match snapshot with user defined", () => {
    const wrapper = shallow(<Navbar {...mockedProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Should match snapshot with undefined user", () => {
    const wrapper = shallow(<Navbar {...mockedProps} user={null} />);

    expect(wrapper).toMatchSnapshot();
  });
});
