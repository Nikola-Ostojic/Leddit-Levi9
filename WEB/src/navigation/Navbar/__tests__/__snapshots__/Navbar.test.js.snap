// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests Should match snapshot with undefined user 1`] = `
<WithStyles(ForwardRef(AppBar))
  className="makeStyles-appBar-1 links"
  position="fixed"
>
  <WithStyles(ForwardRef(Toolbar))>
    <WithStyles(ForwardRef(IconButton))
      aria-label="menu"
      color="inherit"
      edge="start"
      onClick={[Function]}
    >
      <MenuIcon />
    </WithStyles(ForwardRef(IconButton))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-2"
      variant="h6"
    >
      Leddit
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Button))
      color="inherit"
    >
      <NavLink
        to="/auth"
      >
        <WithStyles(ForwardRef(Typography))>
          Login
        </WithStyles(ForwardRef(Typography))>
      </NavLink>
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;

exports[`Navbar tests Should match snapshot with user defined 1`] = `
<WithStyles(ForwardRef(AppBar))
  className="makeStyles-appBar-1 links"
  position="fixed"
>
  <WithStyles(ForwardRef(Toolbar))>
    <WithStyles(ForwardRef(IconButton))
      aria-label="menu"
      color="inherit"
      edge="start"
      onClick={[Function]}
    >
      <MenuIcon />
    </WithStyles(ForwardRef(IconButton))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-2"
      variant="h6"
    >
      Leddit
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Button))
      color="inherit"
    >
      <NavLink
        onClick={[MockFunction]}
        to="/auth"
      >
        <WithStyles(ForwardRef(Typography))>
          Logout 
          user
        </WithStyles(ForwardRef(Typography))>
      </NavLink>
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
